File: 1556-make-two-arrays-equal-by-reversing-subarrays/make-two-arrays-equal-by-reversing-subarrays.py
class Solution:
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        sumtarget = 0
        sumarr = 0

        for i in target:
            sumtarget += (i * i)
        for i in arr:
            sumarr += (i * i)

        return True if sumtarget == sumarr and sum(arr) == sum(target) else False 

File: 1556-make-two-arrays-equal-by-reversing-subarrays/make-two-arrays-equal-by-reversing-subarrays.js
/**
 * @param {number[]} target
 * @param {number[]} arr
 * @return {boolean}
 */
var canBeEqual = function (target, arr) {
    let sortedarr = arr.sort((a, b) => a - b);
    let sortedtarget = target.sort((a, b) => a - b);

    if (sortedtarget.length !== sortedarr.length) {
        return false;
    }

    for (let i = 0; i < arr.length; i++) {
        if (sortedarr[i] !== sortedtarget[i]) {
            return false;
        }
    }

    return true;
};